msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/SliceSparseMatrix.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * (C) Copyright 2015-2017 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+package io.github.msdk.featdet.ADAP3D.common.algorithms;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import io.github.msdk.MSDKException;
+import io.github.msdk.datamodel.rawdata.MsScan;
+import io.github.msdk.datamodel.rawdata.RawDataFile;
+import io.github.msdk.featdet.ADAP3D.datamodel.SparseMatrixTriplet;
+import org.apache.commons.collections.map.MultiKeyMap;
+
+
+public class SliceSparseMatrix {
```

Comment:
Hi Dharak,
Since the code is slowly becoming more complex, could you add a JavaDoc to each new class that explains the basic function and purpose of the class? Thanks!